util/generate_ast/_base.px

```diff
@@ -1,4 +1,9 @@
 class AST(mutable=ast.AST, metaclass=ImmutableMeta):
+    def replace(self, **kwargs):
```

Comment:
One thing to consider is that if `replace` was made a function instead of a method it would work with `ast`, however usage would be slightly more cumbersome.